src/Appwrite/Platform/Workers/Messaging.php

```diff
@@ -245,10 +245,11 @@ private function sendExternalMessage(
                     default => throw new \Exception('Provider with the requested ID is of the incorrect type')
                 };
 
-                $batches = \array_chunk(
-                    \array_keys($identifiersForProvider),
-                    $adapter->getMaxMessagesPerRequest()
-                );
+                $keys = array_keys($identifiersForProvider);
+
+                $batches = $provider->getAttribute('provider') === 'smtp'
+                    ? array_map(fn ($id) => [$id], $keys)
+                    : array_chunk($keys, $adapter->getMaxMessagesPerRequest());
```

Comment:
Let's do a benchmark to compare how long it take to send a message with 1000 recipients with an SMTP provider before and after this change